typescript/infra/scripts/deploy.ts

```diff
@@ -71,9 +72,24 @@ async function main() {
     );
     deployer = new HyperlaneCoreDeployer(multiProvider, ismFactory);
   } else if (module === Modules.HOOK) {
-    throw new Error('Hook deployment unimplemented');
-    // config = envConfig.hooks;
-    // deployer = new HyperlaneHookDeployer(multiProvider);
+    if (!envConfig.hooks) {
+      throw new Error(`No hook config for ${environment}`);
+    }
+    config = envConfig.hooks;
+    const ismFactory = HyperlaneIsmFactory.fromAddressesMap(
+      getAddresses(environment, Modules.ISM_FACTORY),
+      multiProvider,
+    );
+    const mailboxes: ChainMap<Address> = {};
+    for (const chain in getAddresses(environment, Modules.CORE)) {
+      mailboxes[chain] = getAddresses(environment, Modules.CORE)[chain].mailbox;
+    }
```

Comment:
```suggestion
    const mailboxes = getAddresses(environment, Modules.CORE));
```

Reply:
Tried this, get ``` value: {
    mailbox: '0x1..',
    proxyAdmin: '0x2..',
    validatorAnnounce: '0x3...'
  }``` instead of just `mailbox`

Reply:
because your class doesn't expect the object
dont feel strongly here